From 70c60d85db16768db374088343cf14f4effb9d88 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@st.com>
Date: Fri, 29 Jul 2016 17:15:06 +0200
Subject: [PATCH 05/23] v4l2bufferpool: fix condition for copy threshold

Copy threshold should be enabled only if V4L2 available
buffers are smaller than min buffers.
If greater, min buffers is aligned on V4L2 available
buffers.

Change-Id: I8dcba7b69ca832e48e37c4b080c79210f587894a
Reviewed-on: https://gerrit.st.com/56795
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 sys/v4l2/gstv4l2bufferpool.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 3184bd8..17ea7ea 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -743,13 +743,16 @@ gst_v4l2_buffer_pool_start (GstBufferPool * bpool)
        * falling back to copy if the pipeline needed more buffers. This also
        * prevent having to do REQBUFS(N)/REQBUFS(0) everytime configure is
        * called. */
-      if (count != min_buffers || pool->enable_copy_threshold) {
+      if (count < min_buffers || pool->enable_copy_threshold) {
         GST_WARNING_OBJECT (pool,
             "Uncertain or not enough buffers, enabling copy threshold");
         min_buffers = count;
         copy_threshold = min_latency;
       }
 
+      if (count > min_buffers)
+        min_buffers = count;
+
       break;
     }
     case GST_V4L2_IO_USERPTR:
-- 
2.7.4

